src/lib/features/project/project-service.ts

```diff
@@ -304,6 +304,30 @@ export default class ProjectService {
         return featureNaming;
     };
 
+    private validateLinkTemplates = (
+        linkTemplates: IProject['linkTemplates'] | unknown,
+    ): IProject['linkTemplates'] => {
+        if (!linkTemplates || !Array.isArray(linkTemplates)) {
+            throw new BadDataError(
+                'The link templates you provided were invalid.',
+            );
+        }
+
+        for (const linkTemplate of linkTemplates) {
```

Comment:
where do we test validation corner cases?

Reply:
if it's difficult to test via service it can be extracted into validation function and tested in isolation